armi/reactor/tests/test_reactors.py

```diff
@@ -725,6 +725,21 @@ def test_removeAssembliesInRing(self):
             self.assertNotEqual(aLoc[i], a.spatialLocator)
             self.assertEqual(a.spatialLocator.grid, self.r.core.sfp.spatialGrid)
 
+    def test_removeAssembliesInRingByCount(self):
```

Comment:
This test appears to use the same test reactor as `test_removeAssembliesinRingHex`. Since it's a hex reactor, won't the call to `removeAssembliesInRing` run into the same error as the original ticket since the flag isn't being used?

Reply:
I tested this fix outside of ARMI using our proprietary reactor.  But I can't copy that proprietary reactor here, because this is public.  So, without extensive work to produce another test reactor for ARMI, I'm stuck only building unit tests with the test reactor we have.

Reply:
I get this unit test does pass, but I guess I'm not sure why it does. You have a hex reactor object that you are removing a single ring from without the flag. This means that it should be removing assemblies in a circular shape from the test reactor and therefore that the number of rings after removing may not match the number of initial rings minus one. I'm guessing this test passes only because the approximately circular shape of this particular number of rings does actually result in the entire outer ring being removed. 

It seems this test should be testing the resulting shape is actually circular. Something along the lines of removing some number of rings and then checking that given assemblies (that we know are in the circle or not) are removed/not removed. The number of rings chosen to remove for the test should make the resulting reactor not have a uniform number of hex rings to differentiate this behavior from the remove hex rings behavior.